packages/cf.js/src/provider.ts

```diff
@@ -30,8 +33,12 @@ export class Provider {
     return result.appInstances.map(info => new AppInstance(info));
   }
 
-  on(eventName: DappEventType, callback: (e: DappEvent) => void) {
-    // TODO: support notification observers
+  on(eventName: EventType, callback: (e: CounterfactualEvent) => void) {
+    this.eventEmitter.on(eventName, callback);
+  }
+
+  once(eventName: EventType, callback: (e: CounterfactualEvent) => void) {
```

Comment:
What's the common use case for `once`?

I see it's being used in the tests for reject install, but wouldn't you want to constantly listen for reject installs and not just once?

Reply:
*shrug* no idea, but can't hurt to expose a common convenience method.